Data structure for the props

ashwinKumar0505 opened the pull request:
The reason for change of props data structure is, when the workspace gets too big, the updation of props takes some more time. To reduce the time taken, the props data structure is changed.

ashwinKumar0505 committed:
Changed the data structure for the props

ashwinKumar0505 committed:
Fix : bug in deletion of components

ashwinKumar0505 committed:
Refactoring of code

ashwinKumar0505 committed:
Removed component-id from prop object

ashwinKumar0505 committed:
Fix : bugs caused when props datastructure is rewired

sanandnarayan commented:
componentID : [props]
componentID: {propID: prop}

sanandnarayan commented:
before 
`componentID : [props]`
after
`componentID: {propID: prop}`

or
`byID: {propID: {}};
byComponentID: {componentID:[propid1, propid2]}
`


explained here
https://www.loom.com/share/12a24529556542f79c099e150fa61d93

ashwinKumar0505 commented:
I had seen the video bro. I think second approach seems good. I have suggestions, correct me if i am wrong
Can we use prop-name instead of prop-id as keys. Because prop-names will also be unique. There will no duplicate prop-names.
Instead of having byComponentId as another individual property. Can we move it into the component object structure. Each component will have their set of prop-id's. We can fetch the props based on the id's in the component structure.
For Example:

```
Comp-1:{
id:"comp-1",
type:"Button",
parent:"Comp-0",
children:[],
propIds:["prop-1","prop-2"]
}

```
This are just my suggestions

ashwinKumar0505 committed:
Fix : bug caused while rebasing

sanandnarayan commented:
`Props:{
1: {
	byID:{
		prop-KHTASDNFAS : {
			name: “margin”,
			value: “asfasdf,
			derivative’s…
		}
		prop-asfsf {}
	}
	byComponentID:{
		comp-ASFASFDA:[prop-KTSS, ]
	}
}
}
}`

ashwinKumar0505 committed:
Set new types for the props data-structure

ashwinKumar0505 committed:
Modified the reducers based on new prop-datastructure

ashwinKumar0505 committed:
Modified selectors for the new prop data-structure

ashwinKumar0505 committed:
Modified code generation

ashwinKumar0505 committed:
Added prop-id for each and every panels

ashwinKumar0505 committed:
Fix : bugs caused by change in props datastructure

ashwinKumar0505 committed:
Seperate id-prefix for components and props

sanandnarayan commented on src/react-app-env.d.ts:
```diff
@@ -156,10 +156,19 @@ interface IProp {
+
 interface IPropsByComponentId {
-  [componentId: string]: IProp[]
+  [componentId: string]: string[]
```
more stricter

sanandnarayan commented on src/core/models/components/propsOperations.ts:
```diff
@@ -10,43 +10,53 @@ import { ComponentsState } from './components'
+  )
+
+  delete props.byComponentId[propIdIndex]
+  delete props.byId[propId]
```
change to util fn

sanandnarayan commented on src/core/models/components/propsOperations.ts:
```diff
@@ -94,26 +104,36 @@ export const deleteCustomProp = (
+              draftState.customComponentsProps.byComponentId[
+                componentId
+              ].splice(index, 1)
+              delete draftState.customComponentsProps.byId[propId]
```
^

sanandnarayan commented on src/core/models/components/propsExposion.ts:
```diff
@@ -175,24 +142,43 @@ export const unExposeProp = (
+            draftState.componentsById[componentsId],
+            draftState.propsById[propsId],
+          )
+          draftState.componentsById[componentsId] = { ...components }
```
change to util fn to add prop in both variables

ashwinKumar0505 committed:
Refactoring of code

ashwinKumar0505 committed:
Fix : bug while exposing and unexposing prop

ashwinKumar0505 committed:
propname will be shown for custom props

ashwinKumar0505 committed:
Fix: bug in generation of code

ashwinKumar0505 committed:
jsdoc initial configuration

ashwinKumar0505 committed:
Added jsdoc to the required methods and members

ashwinKumar0505 committed:
fix: 🐛 Bugs in moving components in custom component

When the component with exposed prop gets moved into or from the custom
component, the custom props does not gets updated.

ashwinKumar0505 committed:
Merge pull request #44 from Codebrahma/implement-js-docs

Implement Jsdoc
